content/information-distribution/node.md

```diff
@@ -54,14 +57,65 @@ the Matrix room to join.
 Once a Matrix room is joined, client implementations **must** retrieve and save
 the [list of
 aliases](https://matrix.org/docs/spec/client_server/r0.4.0.html#m-room-aliases)
-for this room. In the event of an entry node that's currently being used being
-unreachable, clients implementations and **must** use this list to provide users
-with an updated list of alternative entry nodes and aliases to use. The address
-of the node managing a given alias can be retrieved through a [DNS
-query](https://github.com/matrix-org/synapse/#setting-up-federation). Client
-implementations **can** check for pings in order to filter out the inactive
-nodes from this list.
+for this Matrix room. This can be done by processing all of the `m.room.aliases`
+events in the state of the Matrix room, each of them containing the list of
+aliases defined for a given node. The content for such an event looks like this:
+
+```
+"aliases": [
+    "#alias1:example.com",
+    "#alias2:example.com"
+]
+```
+
+The example above shows the content of the `m.room.aliases` state event for the
+node which uses `example.com` as its server name.
+
+Client implementations **must** process these lists of aliases to generate a map
+mapping all of the entry nodes for this Matrix room with the aliases they
+provide for this room.
+
+### Reaching an entry node
+
+Because the list of aliases only list the nodes' server names, and not their
+FQDN, a node might not be reachable at the address  In order to find the
+effective FQDN and port to reach a node at, client implementations **must**
+perform a DNS query on an `SRV` record for the `_matrix._tcp` sub-domain of the
+server's name which, in the example above, would look like
+`_matrix._tcp.example.com`. If such a record doesn't exist, client
+implementations **shoud** consider that the node is reachable using its server
+name as its FQDN, and the default Matrix encrypted federation port, `8448`, as
+its port.
 
 {{% notice note %}}
-Nodes that aren't entry nodes are named "relay nodes".
+Because most Matrix homeservers can also be reached through the standard `443`
+HTTPS port, client implementations **should** try reaching a node using this
+port instead of the `8448` one, and fall back to the latter port (or any port
+defined in the node's `SRV` DNS record) if closed. This would result in making
+the network traffic generated by Informo client implementations more difficult
+to detect and filter on.
```

Comment:
Just to be sure, the client should:
1. reach `https://example.com:443`
2. if failed reach `https://example.com:8448`
3. if failed do a DNS query at `_matrix._tcp.example.com` and reach the server using the SRV record

Shouldn't the DNS/SRV query be tried first? How does Synapse manage to resolve this? (I couldn't find information about this in Matrix docs)

Reply:
No. What I meant was

1. DNS query
1.1 If 1 succeeds, try replacing the port in the DNS query with `443` to make it less visible in the traffic
1.2 If 1.1 fails, use the port in the DNS response
2. If 1 fails, try to reach the HS's server name on port `443`
3. If 2 fails, try to reach the HS's server name on port `8448`.

I'll rephrase that part in order to make it clearer.

Reply:
Fixed in https://github.com/Informo/specs/pull/2/commits/6c065ccca0dfc382a746fc4d525d730bf1731568.